src/http_server/mod.rs

```diff
@@ -39,9 +44,11 @@ pub async fn run(
     app: impl Bouncer,
     port: u16,
     secret: &[u8],
+    blocked_countries: Vec<String>,
     metrics_provider: impl Fn() -> String + Clone + Send + 'static,
     metrics_port: u16,
     health_provider: impl Fn() -> String + Clone + Send + 'static,
+    geoip_resolver: Option<Arc<MaxMindResolver>>,
```

Comment:
No need to directly depend on impl details here
```suggestion
    geoip_resolver: Option<impl geoip::Resolver>,
```

Reply:
That I can believe, the AWS code is not particularly pretty.

Reply:
What about this suggestion?

Reply:
'Send' and 'Sync' are not on the trait, which is an issue here. Something we can fix in `utils` later on but right now that makes no difference since we only have one resolver.

Reply:
```rust
geoip_resolver: Option<impl geoip::Resolver + Send + Sync>,
````
But you're right, it should probably be on the trait itself